### PR TITLE
Addition of models and service files for  vnsepgdef

### PR DESCRIPTION
**Pull request by shrsr**

### CODE REVIEW

File: client/vnsEPgDef_service.go [hunk 1]
```diff
@@ -0,0 +1,233 @@
+package client
+
+import (
+	"fmt"
+
+	"github.com/ciscoecosystem/aci-go-client/container"
+	"github.com/ciscoecosystem/aci-go-client/models"
+)
+
+func (sm *ServiceManager) CreateEPgDef(name string, legvnode string, node_instance string, l4l7_service_graph_instance_scopeDn string, l4l7_service_graph_instance_graphDn string, l4l7_service_graph_instance_ctrctDn string, tenant string, description string, nameAlias string, vnsEPgDefAttr models.EPgDefAttributes) (*models.EPgDef, error) {
```
Comment (lhercot) [inline client/vnsEPgDef_service.go:10]: scopeDn -> scope_dn , graphDn -> graph_dn and ctrctDn -> ctrct_dn

File: client/vnsEPgDef_service.go [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+	rn := fmt.Sprintf(models.RnvnsEPgDef, name)
+	parentDn := fmt.Sprintf(models.ParentDnvnsEPgDef, tenant, l4l7_service_graph_instance_ctrctDn, l4l7_service_graph_instance_graphDn, l4l7_service_graph_instance_scopeDn, node_instance, legvnode)
+	vnsEPgDef := models.NewEPgDef(rn, parentDn, description, nameAlias, vnsEPgDefAttr)
+	err := sm.Save(vnsEPgDef)
+	return vnsEPgDef, err
+}
+
+func (sm *ServiceManager) ReadEPgDef(name string, legvnode string, node_instance string, l4l7_service_graph_instance_scopeDn string, l4l7_service_graph_instance_graphDn string, l4l7_service_graph_instance_ctrctDn string, tenant string) (*models.EPgDef, error) {
```
Comment (lhercot) [inline client/vnsEPgDef_service.go:18]: scopeDn -> scope_dn , graphDn -> graph_dn and ctrctDn -> ctrct_dn

File: client/vnsEPgDef_service.go (hunk 1 above)
Comment (lhercot) [reply client/vnsEPgDef_service.go:10]: also which DN of the few does this cover?

File: client/vnsEPgDef_service.go [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+	cont, err := sm.Get(dn)
+	if err != nil {
+		return nil, err
+	}
+
+	vnsEPgDef := models.EPgDefFromContainer(cont)
+	return vnsEPgDef, nil
+}
+
+func (sm *ServiceManager) DeleteEPgDef(name string, legvnode string, node_instance string, l4l7_service_graph_instance_scopeDn string, l4l7_service_graph_instance_graphDn string, l4l7_service_graph_instance_ctrctDn string, tenant string) error {
```
Comment (lhercot) [inline client/vnsEPgDef_service.go:30]: scopeDn -> scope_dn , graphDn -> graph_dn and ctrctDn -> ctrct_dn

File: client/vnsEPgDef_service.go [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+	dn := fmt.Sprintf(models.DnvnsEPgDef, tenant, l4l7_service_graph_instance_ctrctDn, l4l7_service_graph_instance_graphDn, l4l7_service_graph_instance_scopeDn, node_instance, legvnode, name)
+	return sm.DeleteByDn(dn, models.VnsepgdefClassName)
+}
+
+func (sm *ServiceManager) UpdateEPgDef(name string, legvnode string, node_instance string, l4l7_service_graph_instance_scopeDn string, l4l7_service_graph_instance_graphDn string, l4l7_service_graph_instance_ctrctDn string, tenant string, description string, nameAlias string, vnsEPgDefAttr models.EPgDefAttributes) (*models.EPgDef, error) {
```
Comment (lhercot) [inline client/vnsEPgDef_service.go:35]: scopeDn -> scope_dn , graphDn -> graph_dn and ctrctDn -> ctrct_dn

File: client/vnsEPgDef_service.go [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+	rn := fmt.Sprintf(models.RnvnsEPgDef, name)
+	parentDn := fmt.Sprintf(models.ParentDnvnsEPgDef, tenant, l4l7_service_graph_instance_ctrctDn, l4l7_service_graph_instance_graphDn, l4l7_service_graph_instance_scopeDn, node_instance, legvnode)
+	vnsEPgDef := models.NewEPgDef(rn, parentDn, description, nameAlias, vnsEPgDefAttr)
+	vnsEPgDef.Status = "modified"
+	err := sm.Save(vnsEPgDef)
+	return vnsEPgDef, err
+}
+
+func (sm *ServiceManager) ListEPgDef(legvnode string, node_instance string, l4l7_service_graph_instance_scopeDn string, l4l7_service_graph_instance_graphDn string, l4l7_service_graph_instance_ctrctDn string, tenant string) ([]*models.EPgDef, error) {
```
Comment (lhercot) [inline client/vnsEPgDef_service.go:44]: scopeDn -> scope_dn , graphDn -> graph_dn and ctrctDn -> ctrct_dn

File: models/vns_epg_def.go [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/ciscoecosystem/aci-go-client/container"
+)
+
+const (
+	DnvnsEPgDef        = "uni/tn-%s/GraphInst_C-[%s]-G-[%s]-S-[%s]/NodeInst-%s/LegVNode-%s/EPgDef-%s"
```
Comment (lhercot) [inline models/vns_epg_def.go:11]: there is actually more than 1 DN so I would havoid using those DnvnsEPgDef and parentDnvns..

**Review by lhercot — CHANGES_REQUESTED** [review]

File: client/vnsEPgDef_service.go (hunk 1 above)
Comment (shrsr) [reply client/vnsEPgDef_service.go]: I took care of this

File: models/vns_epg_def.go (hunk 6 above)
Comment (shrsr) [reply models/vns_epg_def.go]: ok

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**